### ISSUE TITLE
Split the code into more Gradle projects

### ISSUE THREAD

**Issue by tomasbjerre** [wiremock/wiremock#2266]
### Summary

I'm thinking it might be a good idea to split the code into more Gradle projects, still keeping it in this Git repo.

The reasons I see for doing it is:

 1. Allow test code to use Java 17 (and 8 for production code). I think the text block feature would be very nice when working with strings of JSON.
 2. Would perhaps make it easier to understand the code if there were different projects with clear dependencies on each other. Also having names on different parts may help referring to things from documentation.

I think this can be done without breaking change. There can still be (1) the `wiremock` artifact that depends transitively on other artifacts. Or (2) a bunch of artifacts could be bundled into a `wiremock` artifact so that the result is same as today (using shadowJar).

What do you think?

**Comment by oleg-nenashev** [comment]
I would be definitely interested in having the WireMock API interface in a separate library so that I could use it in https://github.com/wiremock/wiremock-testcontainers-java to ensure full WireMock API compatibility

Some other bits like CLI and Client API could be detached too. As additional benefit, it would decrease footprint for API consumers

**Comment by driverpt** [comment]
Throwing in my 2 cents here: I think that it's more valuable right now to split Wiremock into `Server` and `Client`. Test Containers we can easily use a `GenericContainer` to spin it out.

Recently I've found an issue with Micronaut 4 + `@MicronautLambdaTest`, where `micronaut-aws-api-proxy-test` uses a more recent version of Jetty, and it was causing it to assume the same as WireMock's Jetty 9, causing `NoClassDefException`

I managed to fix it by adding the following:
```
    testImplementation(libs.test.wiremock) {
        exclude("org.eclipse.jetty")
        exclude("org.eclipse.jetty.http2")
    }
```

Would be very useful to be able to have a `wiremock-client` with no Jetty Dependencies included

**Comment by oleg-nenashev** [comment]
Taking the gravity of the change, I put it on the roadmap for consideration/review. Needs @tomakehurst's signof as a BDFL before the work starts in this direction

**Comment by mikebell90** [comment]
Frankly I'd like ALL client and server dependencies to be abstracted although this is likely a major issue. For example I don't want apache http client transitive dependencies. I don't want jetty server transitive dependencies. These cause conflicts and also make it difficult to enforce standardized setups.

Ideally the core is completely abstracted from these, and you bind service providers as needed.

**Comment by tomakehurst** [comment]
I agree entirely this is the right way to go. This would also enable a "lite" version of WireMock that implements most of its features but only uses the JDK HTTP client and server.

Unfortunately however some Jetty couplings have crept into a few places in the core, so this will need to be a 4.x+ refactor to cope with the necessary breaking changes.

**Comment by mikebell90** [comment]
I guess I'd go one step further and make JDK client and server bound service providers too. That way everyone fails unless they explicitly bind. Binding multiple providers (for the two types) fails too. Somewhat annoying but leads to clarity.. Otherwise people can get confused and not realize they are falling back on JDK versions.

**Comment by driverpt** [comment]
Suggestion:
* `wiremock-full` (w/ Jetty Http Server + Client)
* `wiremock-lite` (w/ JDK HTTP Client + Server)
* `wiremock-client` (Rest Client only to use with Docker Image + TestContainers)

Or make `wiremock` compatible with MicroProfile and attachable with any MicroProfile-compatible Runtimes available, WDYT ?
